spec/addressable/uri_spec.rb

```diff
@@ -5515,6 +5515,47 @@ def to_s
   end
 end
 
+describe Addressable::URI, "When set the basename " do
+  it "should have the correct basename: 'index.html'" do
+    uri = Addressable::URI.parse("http://test.com")
+    uri.basename = "index.html"
+
+    expect(uri.basename).to eq("index.html")
+    expect(uri.to_s).to eq("http://test.com/index.html")
+  end
+end
+
+describe Addressable::URI, "When change the basename " do
+  it "should have the new basename: 'item'" do
+    uri = Addressable::URI.parse("http://test.com/some/path/index.html")
+
+    expect(uri.basename = "item").to eq("item")
+    expect(uri.to_s).to eq("http://test.com/some/path/item")
+  end
+end
+
+describe Addressable::URI, "After assignment the basename " do
+  it "should have the correct basename: 'subcat_1' "\
+    "when the current basename at the end contains a slash" do
+
+    url = "http://test.com/sections/categories/sub_categories/"
+    uri = Addressable::URI.parse(url)
+
+    expect(uri.basename = "subcat_1").to eq("subcat_1")
+    expect(uri.to_s).to eq("http://test.com/sections/categories/subcat_1")
+  end
+
+  it "should have former path with the new basename: 'category' "\
+     "when the current basename is part of uri path" do
```

Comment:
I think of all your tests, this is the only one that should actually be its own unique `describe` block. However, please match the style of setting up a `describe` block used elsewhere in the test suite, where the URI object is initialized outside the `it` block and set as the test subject.

Reply:
It's probably also a good idea to give a bit more explanation in a comment about why this test exists. It's a good test, particularly as insurance against possible regressions, but it took me awhile to understand why it existed. A clearly written comment would really help here.

Reply:
ok, i got it, i'll write a clarification, thank you